pkg/cmd/admin/policy/review.go

```diff
@@ -0,0 +1,162 @@
+	$  %[1]s -f myresource_with_no_sa.yaml
+	`)
+)
+
+const ReviewRecommendedName = "review"
+
+type reviewOptions struct {
+	client              client.PodSecurityPolicyReviewsNamespacer
+	Namespace           string
+	out                 io.Writer
+	infos               []*resource.Info
+	printer             kubectl.ResourcePrinter
+	FilenameOptions     resource.FilenameOptions
+	ServiceAccountNames []string
+
+	//AllServiceAccounts  bool TODO: defines a way in the API to defines all service accounts
+	approximatePodTemplateForObject func(runtime.Object) (*kapi.PodTemplateSpec, error)
+}
+
+func NewCmdReview(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	o := &reviewOptions{}
+	cmd := &cobra.Command{
+		Use:     name,
+		Short:   "Checks which ServiceAccount can create a Pod",
+		Long:    reviewLong,
+		Example: fmt.Sprintf(reviewExamples, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(f, args, cmd, out))
+			kcmdutil.CheckErr(o.Run())
+		},
+	}
+
+	usage := "Filename, directory, or URL to a file identifying the resource to get from a server."
+	kcmdutil.AddFilenameOptionFlags(cmd, &o.FilenameOptions, usage)
+
+	cmd.Flags().StringSliceVarP(&o.ServiceAccountNames, "serviceaccounts", "s", o.ServiceAccountNames, "List of ServiceAccount names, comma separated")
+	// TODO: defines a way in the API to defines all service accounts: removing omitempty for PodSecurityPolicyReviewSpec.ServiceAccountNames?
+	//cmd.Flags().BoolVar(&o.AllServiceAccounts, "allserviceaccounts", false, "If true, all accessible service accounts will be considered")
+
+	return cmd
+}
+
+func (o *reviewOptions) Complete(f *clientcmd.Factory, args []string, cmd *cobra.Command, out io.Writer) error {
+	if len(args) == 0 && len(o.FilenameOptions.Filenames) == 0 {
+		return kcmdutil.UsageError(cmd, cmd.Use)
+	}
+	namespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.client, _, err = f.Clients()
+	if err != nil {
+		return fmt.Errorf("unable to obtain client: %v", err)
+	}
+	o.Namespace = namespace
+
+	mapper, typer := f.Object()
+	o.infos, err = resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		NamespaceParam(namespace).
+		FilenameParam(enforceNamespace, &o.FilenameOptions).
+		ResourceTypeOrNameArgs(true, args...).
+		ContinueOnError().
+		Flatten().
+		Do().
+		Infos()
+	if err != nil {
+		return err
+	}
+
+	if len(kcmdutil.GetFlagString(cmd, "output")) != 0 {
+		clientConfig, err := f.ClientConfig()
+		if err != nil {
+			return err
+		}
+		version, err := kcmdutil.OutputVersion(cmd, clientConfig.GroupVersion)
+		if err != nil {
+			return err
+		}
+		p, _, err := kcmdutil.PrinterForCommand(cmd)
+		if err != nil {
+			return err
+		}
+		o.printer = kubectl.NewVersionedPrinter(p, kapi.Scheme, version)
```

Comment:
I'm not sure we need this block. If you care about raw output you can always hit the endpoint by yourself. This is a convenience command, that gives you nice output. Similarly to how `oc import-image` works. In that case just go with the describer and you'll be good.